source/Calamari.Tests/KubernetesFixtures/KubernetesContextScriptWrapperLiveFixtureBase.cs

```diff
@@ -89,44 +84,81 @@ CalamariResult ExecuteDirectlyWithScriptEngine(IReadOnlyList<IScriptWrapper> wra
             return new CalamariResult(result.ExitCode, new CaptureCommandInvocationOutputSink());
         }
 
-        protected void DeployWithScriptAndVerifySuccess(Action<TemporaryDirectory> addFilesAction = null)
+        protected void DeployWithDeploymentScriptAndVerifyResult(Func<string, string> addFilesOrPackageFunc = null, bool shouldSucceed = true)
         {
-            SetupTempDirectoryAndVerifyResult(addFilesAction, d =>
+            DeployWithScriptAndVerifyResult(() =>
             {
                 var scriptPath = Path.Combine(testFolder, "KubernetesFixtures/Scripts");
+                var bashScript = File.ReadAllText(Path.Combine(scriptPath, "KubernetesDeployment.sh"));
+                var powershellScript = File.ReadAllText(Path.Combine(scriptPath, "KubernetesDeployment.ps1"));
+                return (bashScript, powershellScript);
+            }, addFilesOrPackageFunc, shouldSucceed);
+        }
+
+        protected void DeployWithKubectlTestScriptAndVerifyResult()
+        {
+            DeployWithScriptAndVerifyResult("#{Octopus.Action.Kubernetes.CustomKubectlExecutable} cluster-info");
+        }
+
+        protected void DeployWithNonKubectlTestScriptAndVerifyResult()
+        {
+            DeployWithScriptAndVerifyResult("echo running target script...");
+        }
+
+        private void DeployWithScriptAndVerifyResult(string script,
+            Func<string, string> addFilesOrPackageFunc = null, bool shouldSucceed = true)
+        {
+            DeployWithScriptAndVerifyResult(() => (script, script), addFilesOrPackageFunc, shouldSucceed);
+        }
+
+        private void DeployWithScriptAndVerifyResult(Func<(string bashScript, string powershellScript)> scriptFactory,
```

Comment:
I think we should just pass in the two scripts as strings here, instead of the function - we're not gaining anything by it that I can see, and it makes it more complex to understand.

Reply:
This is a good point - I've fixed this whole lot of methods and removed the guff. It's a lot neater now.